refactor (my-feed):

1MeghPatel1 opened the pull request:
- Solved MUI Kebab Case Css error
- Refactored useMyFeedPage Hook for removing unnecessary code and dry principle and better naming
- Did some UI fixes in PostCard Footer for PostActionButton

- Styled PostActionButtons as per figma
![image](https://github.com/user-attachments/assets/7bba9a0b-70c7-4fc0-b256-d0b752d49221)

1MeghPatel1 committed:
refactor (my-feed):
- Solved MUI Kebab Case Css error
- Refactored useMyFeedPage Hook for removing unnecessary code and dry principle and better naming
- Did some UI fixes in PostCard Footer for PostActionButton

1MeghPatel1 committed:
Merge branch 'main' into refactor/optimising-my-feed

thekalamanchofficial commented on src/app/(with-sidebar)/myfeed/_hooks/useMyFeedPage.tsx:
```diff
@@ -11,34 +10,33 @@ import { trpc } from "~/server/client";
   setSkip: React.Dispatch<React.SetStateAction<number>>;
   handleLikeButton: (postId: string) => Promise<{ liked: boolean }>;
-  handleChange: (newTab: MyFeedTabsEnum) => void;
+  handleChangeTab: (newTab: MyFeedTabsEnum) => void;
```
handleTabChange

thekalamanchofficial commented on src/app/(with-sidebar)/myfeed/_hooks/useMyFeedPage.tsx:
```diff
@@ -70,184 +68,140 @@ const useMyFeedPage = (): useMyFeedPageReturn => {
+          parentId,
+        });
+
+        setPosts((prevPosts) =>
```
reduce the complexity

thekalamanchofficial commented on src/app/(with-sidebar)/myfeed/_hooks/useMyFeedPage.tsx:
```diff
@@ -70,184 +68,140 @@ const useMyFeedPage = (): useMyFeedPageReturn => {
   useEffect(() => {
-    setPosts((prevPosts) => [...(prevPosts ?? []), ...(postData?.posts ?? [])]);
+    if (postData?.posts) {
+      setPosts((prev) => [...prev, ...postData.posts]);
```
check if we can get rid of this useeffect

thekalamanchofficial commented on src/app/_components/postActionButton/PostActionButton.tsx:
```diff
@@ -1,38 +1,43 @@
         alignItems: "center",
-        minHeight: "auto",
+        minHeight: "26px",
+        minWidth: "88px",
```
match with figma

thekalamanchofficial commented on src/theme.ts:
```diff
@@ -157,10 +157,10 @@ const theme = createTheme({
           },
           "&.MuiChip-root": {
-            "min-height": "32px",
+            "minHeight": "32px",
```
fix it for others also

1MeghPatel1 committed:
fix (Button Styles):
- fixed Buttons styles in theme
- fixed markup to make the side bard Post Footer Section as per the figma

1MeghPatel1 committed:
merge commit

thekalamanchofficial commented on src/app/_components/sidebar/LeftSideBar.tsx:
```diff
@@ -12,35 +12,49 @@ type LeftSideBarProps = {
-        width: "100%",
-        py: 1,
+        minWidth: "264px",
+        minHeight: "1124px",
```
remove the min height

thekalamanchofficial commented on src/app/_components/sidebar/LeftSideBar.tsx:
```diff
@@ -12,35 +12,49 @@ type LeftSideBarProps = {
       sx={{
-        width: "100%",
-        py: 1,
+        minWidth: "264px",
```
remove the min width

thekalamanchofficial commented on src/app/_components/sidebar/LeftSideBar.tsx:
```diff
@@ -12,35 +12,49 @@ type LeftSideBarProps = {
+        minHeight: "1124px",
         backgroundColor: "white",
         position: "relative",
+        paddingBlockEnd: "24px",
```
check if you need it

1MeghPatel1 committed:
revert (Button changes):
- reverted button changes
- changed handleChangeTab to handleTabChange

1MeghPatel1 committed:
merge commit
